### ISSUE TITLE
Duplicate frame preview at keyframes

### ISSUE THREAD

**Issue by dominikWin** [mifi/lossless-cut#1616]
### I have a lot of issues to go through, so in order to make it easier for me to help you, I ask that you please try these things first

- [X] Try with the [newest version from GitHub](https://github.com/mifi/lossless-cut/releases/latest)
- [X] Read the [documentation](https://github.com/mifi/lossless-cut) and [FAQ, Known issues, Troubleshooting](https://github.com/mifi/lossless-cut/blob/master/issues.md)
- [X] Search for your problem under [Issues](https://github.com/mifi/lossless-cut/issues) or [Discussions](https://github.com/mifi/lossless-cut/discussions)
- [X] If you have a problem only with a particular file (or file type), [use this form instead](https://github.com/mifi/lossless-cut/issues/new?assignees=&labels=&template=file-issue.yml)
- [X] Windows 7, 8 and 8.1 are [no longer supported](https://github.com/mifi/lossless-cut/discussions/1476). **This is not a question about Windows 8.1 or older.**

### Operating System

MacOS 13

### Steps to reproduce

1. Open a file in Lossless Cut
3. Move current time a few frames before a keyframe
4. Step one frame at a time forward until you pass the keyframe

Occurs on every file I tried including ones directly output from ffmpeg in h264.

A video where frame 499 is skipped in Lossless Cut:

https://github.com/mifi/lossless-cut/assets/10552362/21afe1b9-c662-411d-800d-e3e6a201189c

### Expected behavior

The preview window should show the frame matching the timelines timestamp.

### Actual behavior

A frame at the key frame is duplicated.

**Comment by mifi** [comment]
See #156 #585

tldr: with html5 video player, it's not possible to seek to a frame number, but to a time value, so losslesscut has implemented the "seek by frame" as a seek to a time (time divided by fps + 1):

https://github.com/mifi/lossless-cut/blob/1fe37ae923c4f2c39d05872cee051a74ff3ae5a8/src/App.jsx#L296

might be something wrong there with the calculation, or maybe your file has irregular/variable fps. Hard to say without having a file that reproduces the problem. I tried with  an mp4/h264 file and I don't see this behavior

**Comment by hqhbaby** [comment]
The same problem, using LosslessCut to cut and combine Youtube's webm format, also has the problem of Duplicate frame preview at keyframes

**Comment by mifi** [comment]
closing in favor of #1216
